x/sophon/types/codec.go

```diff
@@ -0,0 +1,40 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/legacy"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/msgservice"
+)
+
+func RegisterLegacyCodec(cdc *codec.LegacyAmino) {
+	legacy.RegisterAminoMsg(cdc, &MsgRegisterSophon{}, "sedachain/MsgRegisterSophon")
+	legacy.RegisterAminoMsg(cdc, &MsgEditSophon{}, "sedachain/MsgEditSophon")
+	legacy.RegisterAminoMsg(cdc, &MsgTransferOwnership{}, "sedachain/MsgTransferOwnership")
+	legacy.RegisterAminoMsg(cdc, &MsgAcceptOwnership{}, "sedachain/MsgAcceptOwnership")
+	legacy.RegisterAminoMsg(cdc, &MsgCancelOwnershipTransfer{}, "sedachain/MsgCancelOwnershipTransfer")
+	legacy.RegisterAminoMsg(cdc, &MsgAddUser{}, "sedachain/MsgAddUser")
+	legacy.RegisterAminoMsg(cdc, &MsgTopUpUser{}, "sedachain/MsgTopUpUser")
+	legacy.RegisterAminoMsg(cdc, &MsgSettleCredits{}, "sedachain/MsgSettleCredits")
+	legacy.RegisterAminoMsg(cdc, &MsgExpireCredits{}, "sedachain/MsgExpireCredits")
+	legacy.RegisterAminoMsg(cdc, &MsgSubmitReports{}, "sedachain/MsgSubmitReports")
+	legacy.RegisterAminoMsg(cdc, &MsgUpdateParams{}, "sedachain/MsgUpdateParams")
```

Comment:
Not important because we don't use Amino, but this should be named to `cosmos-sdk/x/sophon/MsgUpdateParams` to avoid collision?

Reply:
Turns out amino labels can't be longer than 40 characters so I shortened the prefix to `seda/sophon`....

Reply:
Ah right. And OK looks good to me 👍